src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -725,6 +751,26 @@ private void PackageSourceProvider_PackageSourcesChanged(object sender, EventArg
             UpdateActiveSource(ActivePackageSource);
         }
 
+        private void RestoreEvents_SolutionRestoreCompleted(SolutionRestoredEventArgs args)
+        {
+            _latestRestore = args;
+        }
+
+        private bool ShouldNoOpDueToRestore(SolutionRestoredEventArgs latestRestore)
+        {
+            return latestRestore != null &&
+                   _currentRestore != null &&
```

Comment:
Thought about this, but I'd rather be defensive just in case an event comes in with `null` (not possible with current code).